sdk/rust/src/http.rs

```diff
@@ -28,17 +28,75 @@ pub struct Request {
     body: Vec<u8>,
 }
 
-enum HeaderValue {
+/// A header value.
+///
+/// Since header values do not have to be valid utf8, this allows for
+/// both utf8 strings and bags of bytes.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct HeaderValue {
+    inner: HeaderValueRep,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+enum HeaderValueRep {
+    /// Header value encoded as a utf8 string
     String(String),
+    /// Header value as a bag of bytes
     Bytes(Vec<u8>),
 }
 
 impl HeaderValue {
+    /// Construct a `HeaderValue` from a string
+    pub fn string(str: String) -> HeaderValue {
+        HeaderValue {
+            inner: HeaderValueRep::String(str),
+        }
+    }
+
+    /// Construct a `HeaderValue` from a bag of bytes
+    pub fn bytes(bytes: Vec<u8>) -> HeaderValue {
+        HeaderValue {
+            inner: HeaderValueRep::Bytes(bytes),
+        }
+    }
+
+    /// Get the `HeaderValue` as a utf8 encoded string
+    ///
+    /// Returns `None` if the value is a non utf8 encoded header value
+    pub fn as_str(&self) -> Option<&str> {
+        match &self.inner {
+            HeaderValueRep::String(s) => Some(s),
+            HeaderValueRep::Bytes(_) => None,
```

Comment:
I think it'd be good to try to return a string here, even if the underlying storage is a vec. Probably using
```rust
HeaderValueRep::Bytes(b) => b.from_utf8().ok(),
```

Reply:
Alternatively, that could of course be done when creating the `HeaderValue`, same as what happens inside `into_header_rep` right now